common.mk

```diff
@@ -32,6 +32,26 @@ ifeq ($(and $(LATEST_TAG),$(IS_TAG)),NO)
 endif
 export VERSION
 
+# Try to compute the next version based on the current version using the Punch
+# tool.
+# NOTE: This is a little messy because Punch doesn't support the following at
+# the moment:
+#   - Passing current version as an CLI parameter.
+#   - Outputting the new version to stdout without making modifications to any
+#     files.
+_PUNCH_VERSION_FILE := $(shell mktemp /tmp/oasis-core.XXXXX.py)
```

Comment:
can you make this one lazy too

Reply:
also is there any way to clean this up after we're done with it

Reply:
I've implemented both in #2559.